test/new-e2e/tests/otel/utils/pipelines_utils.go

```diff
@@ -499,11 +494,36 @@ func getLoadBalancingSpans(t assert.TestingT, traces []*aggregator.TracePayload)
 			}
 		}
 	}
-	assert.Equal(t, 12, spans)
+	require.Equal(t, 12, spans)
 	return spanMap
 }
 
-func getLoadBalancingMetrics(t assert.TestingT, metrics []*aggregator.MetricSeries) map[string][]map[string]string {
+func getLoadBalancingLogs(c require.TestingT, s OTelTestSuite, service string) {
+	logs1, err := s.Env().FakeIntake.Client().FilterLogs(service, fakeintake.WithMessageMatching(log1Body))
+	require.NoError(c, err)
+	require.NotEmpty(c, logs1)
+	log1 := logs1[0]
+	log1Tags := getLogTags(c, log1)
+
+	logs2, err := s.Env().FakeIntake.Client().FilterLogs(service, fakeintake.WithMessageMatching(log2Body))
+	require.NoError(c, err)
+	require.NotEmpty(c, logs2)
+	matchedLog := false
+	for _, log2 := range logs2 {
+		// Find second log with the same trace id
+		log2Tags := getLogTags(c, log2)
+		if log1Tags["dd.trace_id"] == log2Tags["dd.trace_id"] {
```

Comment:
I think logs2 can still contains log1 so you may be just getting log1 here, so you may want something like
`if log1.Message != log2.Message && log1Tags["dd.trace_id"] == log2Tags["dd.trace_id"]`